core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -242,7 +246,10 @@ public void close() {
   public CompletableFuture<Message> applyTransaction(TransactionContext trx) {
     try {
       applyJournalEntryCommand(trx);
-      return super.applyTransaction(trx);
+      RaftProtos.LogEntryProto entry = Objects.requireNonNull(trx.getLogEntry());
+      updateLastAppliedTermIndex(entry.getTerm(), entry.getIndex());
+      // no response message is expected
```

Comment:
Can you add more context/details to this comment?

Also, is there a way to add a simple unit test for this behavior?

Reply:
Added a simple test and more comments.